TestLocalBuild error MSB4044

### Current behavior

C:\GitHub\uno\build\uno.ui.build.csproj(117,3): error MSB4044: The "XmlPoke" task was not given a value for the required parameter "Value"

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

- Run VS Command Line
- go to `C:\GitHub\uno\build\`
- Run `TestLocalBuild`

 

### Workaround

_No response_

### Works on UWP/WinUI

_No response_

### Environment

Other

### NuGet package version(s)

Master ff1ce996a141a660bc10e830934be3b5111598fb

### Affected platforms

Build tasks

### IDE

Visual Studio 2022

### IDE version

Visual Studio Version 17.4.0 Preview 6.0

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

--- a/build/TestLocalBuild.CMD
+++ b/build/TestLocalBuild.CMD
@@ -1,15 +1,30 @@
 @echo off
 setlocal ENABLEEXTENSIONS
 set VSLANG=1033
-SET DOTNET_CLI_UI_LANGUAGE = en-US
-chcp 437
-
+SET DOTNET_CLI_UI_LANGUAGE=en-US
+IF NOT DEFINED UNO_UWP_BUILD (SET UNO_UWP_BUILD=true)
+chcp 437 > nul
 if '%Platform%' NEQ '' if '%CI%' NEQ '' goto errorci
 
 if '%Platform%' NEQ '' if '%CI%' == '' goto question
-
 :build
-msbuild uno.ui.build.csproj "/p:CombinedConfiguration=Release|AnyCPU;BUILD_BUILDNUMBER=test_test_8888;GitVersion_FullSemVer=1.0.0.0;GITVERSION_MajorMinorPatch=1.0.0;GITVERSION_CommitsSinceVersionSource=1" /m:16 /clp:PerformanceSummary;Summary /bl
+dotnet-gitversion /version >nul 2>&1
+if %ERRORLEVEL% NEQ 0 (
+ echo "Install gitversion with command: 'dotnet tool install GitVersion.Tool -g'"
+ exit /B 1
+)
+
+for /f %%i in ('dotnet-gitversion /showvariable SemVer') do set GitVersion_SemVer=%%i
+for /f %%i in ('dotnet-gitversion /showvariable MajorMinorPatch') do set GITVERSION_MajorMinorPatch=%%i
+for /f %%i in ('dotnet-gitversion /showvariable CommitsSinceVersionSource') do set GITVERSION_CommitsSinceVersionSource=%%i
+for /f %%i in ('dotnet-gitversion /showvariable VersionSourceSha') do set GITVERSION_VersionSourceSha=%%i
+
+echo "GitVersion_SemVer                   :%GitVersion_SemVer%"
+echo "GITVERSION_MajorMinorPatch          :%GITVERSION_MajorMinorPatch%"
+echo "GITVERSION_CommitsSinceVersionSource:%GITVERSION_CommitsSinceVersionSource%"
+echo "GITVERSION_VersionSourceSha         :%GITVERSION_VersionSourceSha%"
+
+msbuild uno.ui.build.csproj "/p:CombinedConfiguration=Release|AnyCPU;BUILD_BUILDNUMBER=test_test_8888" /m:16 /clp:PerformanceSummary;Summary /bl %*
 pause
 goto end
 :errorci

--- a/build/TestLocalBuild.CMD
+++ b/build/TestLocalBuild.CMD
@@ -2,7 +2,6 @@
 setlocal ENABLEEXTENSIONS
 set VSLANG=1033
 SET DOTNET_CLI_UI_LANGUAGE=en-US
-IF NOT DEFINED UNO_UWP_BUILD (SET UNO_UWP_BUILD=true)
 chcp 437 > nul
 if '%Platform%' NEQ '' if '%CI%' NEQ '' goto errorci
 

--- a/build/Uno.UI.Build.csproj
+++ b/build/Uno.UI.Build.csproj
@@ -49,7 +49,7 @@
 		<CallTarget Targets="UpdateFileVersions;UpdateTasksSHA" Condition="$(_isWindows)" />
 		<CallTarget Targets="PrepareNuGetPackage" />
 	</Target>
-	
+
 	<Target Name="BuildGeneric" AfterTargets="Build" Condition="'$(BuildingInsideVisualStudio)'==''">
 
 		<CallTarget Targets="BuildCI" Condition="'$(Configuration)'=='Release' and $(_isWindows)" />
@@ -60,10 +60,10 @@
 	<Target Name="GeneratePackages" AfterTargets="Build" DependsOnTargets="PrepareBuildAssets" Condition="'$(BuildingInsideVisualStudio)'==''">
 		<CallTarget Targets="BuildNuGetPackage" Condition="'$(Configuration)'=='Release'" />
 	</Target>
-	
+
 	<Target Name="Build2022VSIX" Condition="'$(BuildingInsideVisualStudio)'==''">
 		<CallTarget Targets="UpdateFileVersions" Condition="$(_isWindows)" />
-		
+
 		<MSBuild Properties="Configuration=Release"
              Projects="..\src\SolutionTemplate\UnoSolutionTemplate.VSIX.2022\UnoSolutionTemplate.VSIX.2022.csproj"
              Targets="Restore;Build"
@@ -119,7 +119,7 @@
 							 Value="$(GitVersion_SemVer)"
 							 Namespaces="$(MSBuildDeveloperNamespace)"
 							 />
-		
+
 		<XmlPoke XmlInputPath="%(_legacyProject.Identity)"
 							 Query="//x:PackageReference[@Include='Uno.UI']/@Version"
 							 Value="$(GitVersion_SemVer)"
@@ -206,7 +206,7 @@
 
     <WriteLinesToFile
         File="%(_Sha1Replace.Identity)"
-        Lines="$([System.IO.File]::ReadAllText(%(_Sha1Replace.Identity)).Replace('v0','v$(GITVERSION_VersionSourceSha)'))"
+        Lines="$([System.IO.File]::ReadAllText('%(_Sha1Replace.Identity)').Replace('v0','v$(GITVERSION_VersionSourceSha)'))"
         Overwrite="true" />
 
 	</Target>
@@ -266,7 +266,7 @@
 				 Targets="Restore"
 				 RebaseOutputs="false"
 				 BuildInParallel="true" />
-		
+
 		<MSBuild Properties="Configuration=Release" Projects="..\src\Uno.UWPSyncGenerator\Uno.UWPSyncGenerator.csproj" Targets="Restore;Build" />
 
 		<ItemGroup>
@@ -284,7 +284,7 @@
 	</Target>
 
 	<Target Name="GenerateDoc" DependsOnTargets="BuildSyncGenerator">
-		
+
 		<MSBuild Properties="Configuration=Debug"
 				 Projects="..\src\Uno.Foundation.Logging\Uno.Foundation.Logging.csproj"
 				 Targets="Restore;Build"
@@ -297,7 +297,7 @@
 	</Target>
 
 	<Target Name="RunAPISyncTool" DependsOnTargets="BuildSyncGenerator">
-	
+
 		<Exec Command="..\src\Uno.UWPSyncGenerator\Bin\Release\Uno.UWPSyncGenerator.exe &quot;sync&quot;" />
 	</Target>
 
@@ -305,7 +305,7 @@
 		<Copy SourceFiles="..\src\SolutionTemplate\UnoSolutionTemplate.VSIX.2019\bin\Release\UnoSolutionTemplate.VSIX.vsix"
 					DestinationFiles="$(OutputDir)\vslatest-generic\UnoPlatform-$(GITVERSION_SemVer).x86.vsix" />
 	</Target>
-	
+
 	<Target Name="PublishVSIX2022">
 		<Copy SourceFiles="..\src\SolutionTemplate\UnoSolutionTemplate.VSIX.2022\bin\Release\UnoSolutionTemplate.VSIX.2022.vsix"
 					DestinationFiles="$(OutputDir)\vslatest\UnoPlatform-$(GITVERSION_SemVer).x64.vsix" />
@@ -340,7 +340,7 @@
 
 		<!-- Update Uno.WinUI references version -->
 		<XmlPoke XmlInputPath="%(_NuspecFiles.Identity)" Query="/x:package/x:metadata/x:dependencies//x:dependency[@id='Uno.WinUI']/@version" Value="$(GITVERSION_SemVer)" Namespaces="$(NugetNamespace)" />
-		
+
 		<!-- Update Uno.Foundation.Runtime.WebAssembly references version -->
 		<XmlPoke XmlInputPath="%(_NuspecFiles.Identity)" Query="/x:package/x:metadata/x:dependencies//x:dependency[@id='Uno.Foundation.Runtime.WebAssembly']/@version" Value="$(GITVERSION_SemVer)" Namespaces="$(NugetNamespace)" />
 
@@ -462,16 +462,16 @@
 
 		<!-- Lottie move file -->
 		<Move SourceFiles="..\src\AddIns\Uno.UI.Lottie\buildTransitive\Uno.UI.Lottie.targets" DestinationFiles="..\src\AddIns\Uno.UI.Lottie\buildTransitive\$(PackageNamePrefix).Lottie.targets" />
-		
+
 		<!-- MSAL move file -->
 		<Move SourceFiles="..\src\AddIns\Uno.UI.MSAL\buildTransitive\Uno.UI.MSAL.targets" DestinationFiles="..\src\AddIns\Uno.UI.MSAL\buildTransitive\$(PackageNamePrefix).MSAL.targets" />
-		
+
 		<!-- Svg move file -->
 		<Move SourceFiles="..\src\AddIns\Uno.UI.Svg\buildTransitive\Uno.UI.Svg.targets" DestinationFiles="..\src\AddIns\Uno.UI.Svg\buildTransitive\$(PackageNamePrefix).Svg.targets" />
-		
+
 		<!-- XamlHost move file -->
 		<Move SourceFiles="..\src\Uno.UI.XamlHost\buildTransitive\Uno.UI.XamlHost.props" DestinationFiles="..\src\Uno.UI.XamlHost\buildTransitive\$(PackageNamePrefix).XamlHost.props" />
-		
+
 		<XmlPoke XmlInputPath=".\Uno.WinUI.XamlHost.nuspec" Query="/x:package/x:files/x:file[@src='..\src\Uno.UI.XamlHost\buildTransitive\Uno.WinUI.XamlHost.props']/@src" Value="..\src\Uno.UI.XamlHost\buildTransitive\$(PackageNamePrefix).XamlHost.props" Namespaces="$(NugetNamespace)" Condition="'$(UNO_UWP_BUILD)'=='true'"/>
 
 		<!-- Runtime package move files -->
@@ -527,7 +527,7 @@
 		<Exec Command="dotnet tool install --tool-path $(MSBuildThisFileDirectory)\tools Uno.PackageDiff --version 1.1.0-dev.14" IgnoreExitCode="true" />
 		<Exec Command="$(MSBuildThisFileDirectory)\tools\generatepkgdiff.exe --base=$(PackageNamePrefix) --other=$(PackageNamePrefix).$(GITVERSION_SemVer).nupkg --diffignore=PackageDiffIgnore.xml --outfile=$(OutputDir)\ApiDiff.$(GITVERSION_SemVer).md" />
 	</Target>
-	
+
 	<Target Name="ValidatePackageReferenceAPI" AfterTargets="BuildNuGetPackage">
 		<PropertyGroup>
 			<PackageNamePrefix>Uno.WinUI</PackageNamePrefix>

